force-app/main/default/classes/StaticTask.cls

```diff
@@ -1,38 +1,106 @@
-        }
-        if (element.contains('LastName')) {
-          parsedContact.LastName = element.split(':')[1].replaceAll(' ', '');
-        }
-        if (element.contains('Email')) {
-          if (String.isNotBlank(element.split(':')[1])) {
-            parsedContact.Email = element.split(':')[1].replaceAll(' ', '');
-            emailToContactMap.put(
-              element.split(':')[1].replaceAll(' ', ''),
-              parsedContact
-            );
-          }
-        }
+
+  public static Map<String, Contact> extractEmailToContactMap (String payload) {
+
+    List<String> payloadAsStringList = StaticTask.splitPayloadByElementStringList(payload);
+    Map<Integer, List<String>> sObjectFieldsMap = StaticTask.splitStringListByObjectFieldMap(payloadAsStringList);
+    List<Contact> contactList = StaticTask.splitObjectFieldMapByRecords(sObjectFieldsMap);
+    Map<String, Contact> emailAndContactMap = StaticTask.createEmailAndContactMap(contactList);
+
+    return emailAndContactMap;
+  }
+
+
+  public static List<String> splitPayloadByElementStringList (String payload) {
+    return payload.split(';');
+  }
+
+  
+  public static Map<Integer, List<String>> splitStringListByObjectFieldMap (List<String> stringList) {
+    
+    Map<Integer, List<String>> sObjectFieldMap = new Map<Integer, List<String>>();
+    
+    for (Integer i = 0; i < stringList.size(); i++) {
+      sObjectFieldMap.put(i, stringList[i].split(','));
+    }
+
+    return sObjectFieldMap;
+  }
+
+
+  public static List<Contact> splitObjectFieldMapByRecords (Map<Integer, List<String>> integerStringListMap) {
+    
+    List<Contact> contactList = new List<Contact>();
+
+    for (Integer i = 0; i < integerStringListMap.size(); i++) {
+      List<String> singleListFromMap = integerStringListMap.get(i);
+      Contact record = StaticTask.formContactFromStringList(singleListFromMap);
+      contactList.add(record);
+    }
+
+    return contactList;
+  }
+
+
+  public static Map<String, Contact> createEmailAndContactMap (List<Contact> contactList) {
+
+    Map<String, Contact> emailAndContactMap = new Map<String, Contact>();
+
+    for (Contact record : contactList) {
+      String email = (String) record.get('Email');
+
+      if (email != null) {
+        emailAndContactMap.put(email, record);
       }
     }
-    return emailToContactMap;
+
+    return emailAndContactMap;
+  }
+
+
+  public static String getValue (String pair) {
+    return pair.substringAfter(':').trim();
+  }
+
+
+  public static String getField (String pair) {
+    return pair.substringBefore(':').trim();
+  }
+
+
+  public static Boolean isValueBlank (String pair) {
+    return String.isBlank(pair.substringAfter(':'));
+  }
+
+
+  public static Contact formContactFromStringList (List<String> listOfContactFields) {
+    
+    Contact record = new Contact();
+
+    for (String fieldString : listOfContactFields) {
+      if (StaticTask.isValueBlank(fieldString)) {
```

Comment:
nice. I like how you split `isValueBlank` and `getValue`. This leaves a room for refactoring, cause the realization of `isValueBlank` is hidden from the caller